Data/ByteString/Lazy.hs

```diff
@@ -538,70 +538,30 @@ compareLength (Chunk c cs) toCmp  = compareLength cs (toCmp - fromIntegral (S.le
 {-# RULES 
 "ByteString.Lazy length/compareN -> compareLength" [~1] forall t n.
   compare (length t) n = compareLength t n
-  #-}
-
-{-# RULES 
 "ByteString.Lazy compareN/length -> compareLength" [~1] forall t n.
-  compare n (length t) = negateOrdering $ compareLength t n
-  #-}
-
-{-# RULES 
+  compare n (length t) = compare EQ $ compareLength t n
```

Comment:
This is not a readability improvement IMHO, but I don't care too much.

Reply:
A comment could be added that `compare EQ LT = GT` and vice versa.  Would that be helpful?  I just didn't see a need to define a new function for this...

Reply:
Yeah, that comment would be helpful. :+1:

Reply:
Done.